src/Uno.UI/UI/Xaml/Controls/Grid/Grid.Layout.cs

```diff
@@ -407,18 +408,111 @@ private void LayoutChildren(Span<DoubleRange> calculatedPixelColumns, Span<Doubl
 				}
 			}
 
+			// Auto columns should take up the larger of their measured width and their MinWidth
```

Comment:
You say "Auto columns", but it seems like the calculation is applied on all columns.

Reply:
there's an `IsAuto` check